packages/js/src/ui/context/LocalizationContext.tsx

```diff
@@ -1,17 +1,47 @@
 import { Accessor, createContext, createMemo, ParentProps, useContext } from 'solid-js';
-import { defaultLocalization } from '../config/defaultLocalization';
-import type { Localization, LocalizationKey } from '../types';
+import { defaultLocalization, dynamicLocalization } from '../config/defaultLocalization';
 
-type LocalizationValue<K extends LocalizationKey> = (typeof defaultLocalization)[K];
-type LocalizationParams<K extends LocalizationKey> = LocalizationValue<K> extends (args: infer P) => any
-  ? P
+type DefaultLocalizationKey = keyof typeof defaultLocalization;
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type LocalizationKey = DefaultLocalizationKey;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** do we need duplicated type? maybe just call `DefaultLocalizationKey --> LocalizationKey` ?

Reply:
I'll change it. This is leftover because I was trying to get it working with custom strings as well.